File: 01_Introduction/01_maxValue/maxValue.py
# My Solution
def max_value(nums):
  max = nums[0]
  for n in nums:
    print(n)
    if n > max:
      max = n
  return max

max_value([4, 7, 2, 8, 10, 9])

# Alvin's Solution

def max_value(nums):
  maximum = float('-inf')
  
  for num in nums:
    if num > maximum:
      maximum = num
      
  return maximum

# n = length of list
# Time: O(n)
# Space: O(1)

File: Structy_1/01_Introduction/01_maxValue/maxValue.js
// My Solution

// const maxValue = (nums) => {
//   let max = nums[0];
//   for (let i = 1; i < nums.length; i++) {
//     if (nums[i] > max) {
//       max = nums[i];
//     }
//   }
//   return max;
// };

// Alvin's Solution

const maxValue = (nums) => {
  let maximum = -Infinity;

  for (let num of nums) {
    // iterates through number, not the indices
    if (num > maximum) maximum = num;
  }

  return maximum;
};

console.log(maxValue([1,2,3,4,5,]))

// n = length of array
// Time: O(n)
// Space: O(1)
